packages/functions/src/splice-node.ts

```diff
@@ -0,0 +1,28 @@
+import { Scene, Node } from '@gltf-transform/core';
+
+/**
+ * Create a new node as child of the target {@link Scene} or {@link Node}.
+ * Its existing children will be reparented under the new node.
+ *
+ * Example:
+ *
+ * ```javascript
+ * import { transformScene } from '@gltf-transform/functions';
+ *
+ * let scene = document.getRoot().getDefaultScene()
+ * // rotate the entire scene a quarter turn around the y axis
+ * spliceNode(scene).setRotation([0, Math.SQRT1_2, 0, Math.SQRT1_2]);
+ * ```
+ *
+ * @param parent Node or Scene whose children to reroot
+ * @returns The new parent node
+ */
+export function spliceNode(parent: Scene | Node): Node {
+	const newNode = new Node(parent.getGraph());
+	for (const child of parent.listChildren()) {
+		if (child.getSkin()) continue;
```

Comment:
Thinking more about my recommendation from https://github.com/donmccurdy/glTF-Transform/issues/1530#issuecomment-2414064338 here, and unfortunately it's not as simple as I'd been hoping. 

1. As mentioned, we will get validation warnings if we put a parent above a skinned mesh. Technically it's valid, these are just 'hints' or 'warnings' at the validator level, because the transform won't do anything to the skinned mesh _directly_, transforms affect only the joints.
2. However, it's possible that the skinned mesh's joints are descendants of the node. In that case, skipping the skinned mesh will cause the transform to not affect the joints as expected, and thus not to affect the skinned mesh.

Probably (2) is the more critical error, as it will change the rendered result.

We could search the child's subtree for any joints, and skip the skinned mesh only if no joints are found. But this might be a more complex and less predictable operation than users are expecting, calling a method like 'spliceNode'. If we skip that special case, then the method body becomes something more like:

```javascript
const node = Document.fromGraph(parent.getGraph()).createNode();
for (const child of parent.listChildren()) {
  node.addChild(child);
}
parent.addChild(newNode);
```